File: go/src/leetcode/offer/Offer18.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/28 23:36
 */

func deleteNode(head *ListNode, val int) *ListNode {
	dummy := &ListNode{}
	dummy.Next = head
	pre := dummy
	cur := pre.Next
	for cur != nil && cur.Val != val {
		cur = cur.Next
		pre = pre.Next
	}
	pre.Next = cur.Next
	return dummy.Next
}

File: java/src/leetcode/offer/Offer18.java
package leetcode.offer;

import common.ListNode;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/12 21:34
 */
public class Offer18 {

    public ListNode deleteNode(ListNode head, int val) {
        if (head == null) {
            return null;
        }
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode cur = dummy;
        while (cur != null && cur.next != null) {
            if (cur.next.val == val) {
                cur.next = cur.next.next;
            }
            cur = cur.next;
        }
        return dummy.next;
    }
}
